x/nft/client/cli/query_test.go

```diff
@@ -0,0 +1,383 @@
+)
+
+func (s *CLITestSuite) TestQueryClass() {
+	testCases := []struct {
+		name string
+		args struct {
+			ClassID string
+		}
+		expectErr bool
+	}{
+		{
+			name: "class id exist",
+			args: struct {
+				ClassID string
+			}{
+				ClassID: testClassID,
+			},
+			expectErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			cmd := cli.GetCmdQueryClass()
+			var args []string
+			args = append(args, tc.args.ClassID)
+			args = append(args, fmt.Sprintf("--%s=json", tmcli.OutputFlag))
+			out, err := clitestutil.ExecTestCLICmd(s.clientCtx, cmd, args)
+
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				var result nft.QueryClassResponse
+				err = s.clientCtx.Codec.UnmarshalJSON(out.Bytes(), &result)
+				s.Require().NoError(err)
+			}
+		})
+	}
+}
+
+func (s *CLITestSuite) TestQueryClasses() {
+	testCases := []struct {
+		name      string
+		expectErr bool
+	}{
+		{
+			name:      "no params",
+			expectErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			cmd := cli.GetCmdQueryClasses()
+			var args []string
+			args = append(args, fmt.Sprintf("--%s=json", tmcli.OutputFlag))
+			out, err := clitestutil.ExecTestCLICmd(s.clientCtx, cmd, args)
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				var result nft.QueryClassesResponse
+				err = s.clientCtx.Codec.UnmarshalJSON(out.Bytes(), &result)
+				s.Require().NoError(err)
+			}
+		})
+	}
+}
+
+func (s *CLITestSuite) TestQueryNFT() {
+	testCases := []struct {
+		name string
+		args struct {
+			ClassID string
+			ID      string
+		}
+		expectErr bool
+	}{
+		{
+			name: "exist nft",
+			args: struct {
+				ClassID string
+				ID      string
+			}{
+				ClassID: testClassID,
+				ID:      testID,
+			},
+			expectErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			cmd := cli.GetCmdQueryNFT()
+			var args []string
+			args = append(args, tc.args.ClassID)
+			args = append(args, tc.args.ID)
+			args = append(args, fmt.Sprintf("--%s=json", tmcli.OutputFlag))
+
+			out, err := clitestutil.ExecTestCLICmd(s.clientCtx, cmd, args)
+			if tc.expectErr {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+				var result nft.QueryNFTResponse
+				err = s.clientCtx.Codec.UnmarshalJSON(out.Bytes(), &result)
+				s.Require().NoError(err)
+			}
+		})
+	}
+}
+
+func (s *CLITestSuite) TestQueryNFTs() {
+	accounts := testutil.CreateKeyringAccounts(s.T(), s.kr, 1)
+
+	testCases := []struct {
+		name string
+		args struct {
+			ClassID string
+			Owner   string
+		}
+		expectErr    bool
+		expectResult []*nft.NFT
+	}{
+		{
+			name: "class id does not exist",
+			args: struct {
+				ClassID string
+				Owner   string
+			}{
+				ClassID: "class",
+				Owner:   accounts[0].Address.String(),
+			},
+			expectErr:    false,
+			expectResult: []*nft.NFT{},
+		},
```

Comment:
are these type of tests needed? since they rely on keeper logic, and `expectErr` also set `false` generally it should be true.

Reply:
thanks for pointing out @atheeshp, necessary changes are done now